sdk/eventhub/azure-eventhubs/azure/eventhub/aio/_producer_async.py

```diff
@@ -95,33 +100,35 @@ def _create_handler(self):
             keep_alive_interval=self._keep_alive,
             client_name=self._name,
             link_properties=self._link_properties,
-            properties=self._client._create_properties(  # pylint: disable=protected-access
-                self._client._config.user_agent),  # pylint:disable=protected-access
+            properties=create_properties(self._client._config.user_agent),  # pylint:disable=protected-access
             loop=self._loop)
 
     async def _open_with_retry(self):
         return await self._do_retryable_operation(self._open, operation_need_param=False)
 
-    async def _send_event_data(self, timeout_time=None, last_exception=None):
-        if self._unsent_events:
-            await self._open()
+    def _set_msg_timeout(self, timeout_time, last_exception):
+        if timeout_time:
```

Comment:
Maybe personal preference, but I generally favor reducing indentation level in cases like this, by returning early:
```
if not foo:
    return 

# block of code
```
rather than having the entire function body inside a conditional.